runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/stream/HttpServerFactory.java

```diff
@@ -4847,11 +4877,23 @@ else if (!isCorsRequestAllowed(binding, headers))
                                     .headers(hs -> headers.forEach((n, v) -> hs.item(h -> h.name(n).value(v))))
                                     .build();
 
-                            exchange.doRequestBegin(traceId, beginEx);
+                            HttpRequestType request = binding.resolveRequest(headers::get);
+
+                            final Http2Exchange exchange = new Http2Exchange(originId, routedId, NO_REQUEST_ID, streamId,
+                                exchangeAuth, traceId, policy, origin, contentLength, request);
 
-                            if (endRequest)
+                            boolean isValid = binding.validateHeader(request, beginEx);
```

Comment:
This should be called `validateHeaders` (plural).

Reply:
done

Reply:
done